test/test-examples-server-https-support-extended.js

```diff
@@ -0,0 +1,183 @@
+        server_root: 'examples'
+      });
+
+      return request(testServer.express)
+        .get('/alexa/helloworld')
+        .expect(200).then(function(response) {
+          // First, check that the actual server instances exist or not
+          expect(testServer.httpInstance).to.exist;
+          expect(testServer.httpsInstance).to.not.exist;
+
+          // Then, check if the ports are actually bound or not
+          isPortTaken({ port: 3000 }, function(err, result) {
+            expect(err).to.not.exist;
+            expect(result).to.be.true;
+
+            isPortTaken({ port: 6000 }, function(err, result) {
+              expect(err).to.not.exist;
+              expect(result).to.be.false;
+            });
+          });
+        });
+    });
+
+    it("should have the HTTP and HTTPS server instances running", function() {
+      testServer = alexaAppServer.start({
+        port: 3000,
+        server_root: 'examples',
+        httpsEnabled: true,
+        httpsPort: 6000,
+        privateKey: 'private-key.pem',
+        certificate: 'cert.cer',
+        chain: 'cert.ca_bundle',
+        passphrase: "test123"
+      });
+
+      return request(testServer.express)
+        .get('/alexa/helloworld')
+        .expect(200).then(function(response) {
+          // First, check that the actual server instances exist or not
+          expect(testServer.httpInstance).to.exist;
+          expect(testServer.httpsInstance).to.exist;
+
+          // Then, check if the ports are actually bound or not
+          isPortTaken({ port: 3000 }, function(err, result) {
+            expect(err).to.not.exist;
+            expect(result).to.be.true;
+
+            isPortTaken({ port: 3000 }, function(err, result) {
+              expect(err).to.not.exist;
+              expect(result).to.be.true;
+            });
+          });
+        });
+    });
+  });
+
+  describe("Specific address given (127.0.0.1)", function() {
+    it("should only have the HTTP server instance running", function() {
+      testServer = alexaAppServer.start({
+        port: 3000,
+        host: '127.0.0.1',
+        server_root: 'examples'
+      });
+
+      return request(testServer.express)
+        .get('/alexa/helloworld')
+        .expect(200).then(function(response) {
+          // First, check that the actual server instances exist or not
+          expect(testServer.httpInstance).to.exist;
+          expect(testServer.httpsInstance).to.not.exist;
+
+          // Then, check if the ports are actually bound or not
+          isPortTaken({ port: 3000, host: '127.0.0.1' }, function(err, result) {
+            expect(err).to.not.exist;
+            expect(result).to.be.true;
+
+            isPortTaken({ port: 6000, host: '127.0.0.1' }, function(err, result) {
+              expect(err).to.not.exist;
+              expect(result).to.be.false;
+            });
+          });
+        });
+    });
+
+    it("should have the HTTP and HTTPS server instances running", function() {
```

Comment:
Well, we need all the ideas that we can get. ;) I do agree on fixing the bug, although as I've said, it should be obvious to the developer. If we were to go with this route, some other changes may need to be employed.

1. The `httpsPort` is not needed since that should be handled by the `port` flag.
2. The `httpsEnabled` flag should be either documented in that only https will be enabled, and/or the flag name itself should be changed.

Personally, I don't mind any particular fix, so long as it makes using the library a bit easier in some aspect. @mreinstein thoughts on this?